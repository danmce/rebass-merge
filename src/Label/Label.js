import React from 'react'
import PropTypes from 'prop-types'
import { Label as RebassLabel } from '@rebass/forms'

const Label = ({ children, htmlFor }) => (
  <RebassLabel htmlFor={htmlFor}>{children}</RebassLabel>
)

Label.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  htmlFor: PropTypes.string
}

export default Label
