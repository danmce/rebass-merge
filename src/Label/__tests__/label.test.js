import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Label from '../Label'

test('renders Button correctly', () => {
  const label = <Label htmlFor='email'>Email</Label>

  expect(toJson(enzyme.mount(label))).toMatchSnapshot('enzyme.mount')
})
