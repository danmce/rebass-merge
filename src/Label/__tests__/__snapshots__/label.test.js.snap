// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button correctly: enzyme.mount 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<Label
  htmlFor="email"
>
  <ForwardRef
    htmlFor="email"
  >
    <Styled(div)
      __css={
        Object {
          "width": "100%",
        }
      }
      as="label"
      htmlFor="email"
      tx="forms"
      variant="label"
    >
      <label
        className="emotion-0"
        htmlFor="email"
      >
        Email
      </label>
    </Styled(div)>
  </ForwardRef>
</Label>
`;
